File: Develop/Backend/src/main/java/com/hientran/sohebox/sco/AccountSCO.java
package com.hientran.sohebox.sco;

import com.fasterxml.jackson.annotation.JsonInclude;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;
import lombok.ToString;

@Data
@ToString
@AllArgsConstructor
@NoArgsConstructor
@JsonInclude(JsonInclude.Include.NON_NULL)
@EqualsAndHashCode(callSuper = true)
public class AccountSCO extends BaseSCO {
	private SearchNumberVO id;
	private SearchNumberVO user;
	private SearchNumberVO accountType;
	private SearchTextVO accountName;
	private SearchTextVO note;

	// Sub Field //
	private SearchTextVO userName;
	private SearchTextVO accountTypeName;
}

File: Develop/Frontend/src/app/scos/accountSCO.ts
import { BaseSCO } from './baseSCO';
import { SearchNumber } from './core_sco/searchNumber';
import { SearchText } from './core_sco/searchText';

export class AccountSCO extends BaseSCO {
  id: SearchNumber;
  user: SearchNumber;
  accountType: SearchNumber;
  accountName: SearchText;
  note: SearchText;
  userName: SearchText;
  accountTypeName: SearchText;
}
